node/src/indexer/tx_sender.rs

```diff
@@ -20,31 +24,33 @@ async fn submit_tx(
     indexer_state: Arc<IndexerState>,
     method: String,
     params_ser: String,
-) -> Option<Nonce> {
+    gas: Gas,
+) -> bool {
```

Comment:
hmm, what do you think of turning this into an `anyhow::Result<()>`? Benefits are (1) it's clear what true/false each means; (2) the caller cannot inadvertently ignore the result without triggering a compiler warning; (3) you can tracing::error once at the call site.